src/check/name/stringname/mod.rs

```diff
@@ -35,8 +33,26 @@ impl Display for StringName {
     }
 }
 
+impl CollectionType for StringName {
+    fn collection_type(&self, ctx: &Context) -> Option<Name> {
```

Comment:
Need to think about this method. What should it return?
Ideally we wan't to go beyond collections of just integers, but it should be anything.

Reply:
See comment on Environemnt and unifcation stage below.
It should just take a Name argument, and assume that we want a `Collection`.
This should work properly as `Collection` is a parent of `List` (which is what we have as palceholder type).